### PR TITLE
Prepare for SettingDSL

### PR DESCRIPTION
**Pull request by ezr-ondrej**
we use setting class quite a lot, first I want to move the heavy lifting to `SettingRegistry`.
Then we can move to implementing the setting definitions in the DSL

### CODE REVIEW

File: lib/foreman_templates.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 require 'foreman_templates/engine'
 
 module ForemanTemplates
+  BASE_SETTING_NAMES = %w(repo branch dirname filter negate)
+  COMMON_SETTING_NAMES = BASE_SETTING_NAMES | %w(verbose)
```
Comment (adamruzicka) [inline lib/foreman_templates.rb:5]: Where is this used?

File: app/views/template_sync_settings/show.json.rabl [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 node do |setting|
-  { :name => setting.short_name }
+  { :name => setting.name.delete_prefix('template_sync_') }
 end
 
 attributes :id, :value, :description, :settings_type, :default, :full_name
 
 node do |setting|
-  { :selection => setting.selection }
+  { :selection => (setting.select_values&.map { |key, label| { value: key, label: label } } || []) }
```
Comment (adamruzicka) [inline app/views/template_sync_settings/show.json.rabl:10]: In general I prefer providing default values as soon as possible, to me it is easier to read.
```
(setting.select_values || []).map { |k, v| ... }
```

As this is a style-only change, feel free to ignore this comment

**Review by adamruzicka — COMMENTED** [review]
Also I'm getting this when opening the form. It doesn't seem to happen with master

image

File: lib/foreman_templates.rb (hunk 1 above)
Comment (ezr-ondrej) [reply lib/foreman_templates.rb:5]: nowhere, I just felt to have the common_settings full list. but I can do without.

**Comment by ezr-ondrej** [comment]
The error was fixed in `master`, I only had an old base for this PR. It should work now.

**Comment by adamruzicka** [comment]
In general +1, but test failures are most likely related

**Comment by ezr-ondrej** [comment]
Indead, we need to reload values in the SettingRegistry to reflect the changes.

**Comment by ezr-ondrej** [comment]
:green_apple:

**Review by adamruzicka — APPROVED** [review]
LGTM, but I don't have merge permissions

**Pull request merged**